python_build_requirements.txt

```diff
@@ -2,7 +2,7 @@ contextlib2==21.6.0
 Mako==1.2.2
 MarkupSafe==2.0.1
 schema==0.7.4
-black>=21.10b0
+black==23.3.0
```

Comment:
This is to avoid getting an alpha version installed that includes proposed formatting changes for 2024. Setting it to the same version used in the [ni-python-styleguide](https://github.com/ni/python-styleguide/blob/cc47dab3ee0b56be594f956834c5e29d5124e4b9/poetry.lock#L3).